lib/bn.js

```diff
@@ -52,6 +52,7 @@ else
 
 BN.BN = BN;
 BN.wordSize = 26;
+BN._hasBuffer = typeof Buffer !== 'undefined';
```

Comment:
Pushed an implementation based on the suggested. It seems to work with browserify, the tests and an actual web browser.